### DISCUSSION TITLE
improve container tags

### DISCUSSION THREAD

**Discussion by hadmut** [voxpupuli/crafty#22] [Ideas]
Hi, 
just a proposal: 

I saw that there are tags 
["7.13.0","main-8","latest","main-7","v1.0.0-7","v1.0.0-8"]

but none of them is really explained in the README.md. I found some description in 
https://github.com/voxpupuli/crafty/blob/main/docs/how-to-release.md

but it is rather confusing, because is says "don't use main-7, it can break any time", while "v1.0.0-7" is difficult to use, because it can change it's version number any time. Furthermore, there's no way to tell from the tags, that the containers are actually using 7.14.0 and 8.3.0 in build_versions.json.

My proposal is to follow current best practices and 

- have tags like  latest-7 , latest-8 , and latest, always pointing to the usable and recommended containers for production environments

- use v1.0.0-7 or main-7 as a development tag and not for productive use, because that v1.0.0 version number and branch name is important for your developers, but that tag should not keep changing all the time for production use. 

- explain all these at prominent location in README.md

regards

**Comment by tuxmea** [comment]
We already dicussed several options on how to ensure that changes within the container (not a new puppetxxx version) can be differentiated.
Consider that one adds a new ENV var to configure something inside the container.
e.g. enable/disable environment delete API endpoint.

Therefor we decided to have a normal version describing the functional basis and append the puppet major version which is used inside the container.

Yes, I agree that the option to have the full puppet version is desired.

@rwaffen @bastelfreak we should have another discussion on this.

Regarding `main` and `latest`:

At the moment we build from every commit to `main` brach. The tag `main` indicates that thinkgs might not have been fully tested.

How about the following:

- commits to main build `main` tag
- tags will be built using tag name
- verified and tested tags should receive the tag `latest` which should be built as `latest`
  - this means that `latest`is always a moving tag. Do we want this?

**Comment by rwaffen** [comment]
to find out atm what is in a tag one can see this like this: https://github.com/voxpupuli/container-puppetserver/blob/v1.1.1/build_versions.json

we will work to make this more visible in the next releases,

**Comment by rwaffen** [comment]
The container tag `latest` is moving constantly. it is automatically added on each build. and the last container which is build keeps it until the next build happens.

**Comment by tuxmea** [comment]
Proposal:

`<puppet_version>-<container version>`

e.g.

`7.13.0-1.1.2`
or `8.2.3-1.0.1`

**Comment by rwaffen** [comment]
i will move this issue to crafty and convert it into a discussion

**Comment by rwaffen** [comment]
newest changes see https://github.com/voxpupuli/crafty/pull/25
